src/privatesend.cpp

```diff
@@ -300,69 +367,18 @@ std::string CPrivateSend::GetMessageByID(PoolMessage nMessageID)
     }
 }
 
-bool CDarksendQueue::Sign()
-{
-    if(!fMasterNode) return false;
-
-    std::string strMessage = vin.ToString() + boost::lexical_cast<std::string>(nDenom) + boost::lexical_cast<std::string>(nTime) + boost::lexical_cast<std::string>(fReady);
-
-    if(!CMessageSigner::SignMessage(strMessage, vchSig, activeMasternode.keyMasternode)) {
-        LogPrintf("CDarksendQueue::Sign -- SignMessage() failed, %s\n", ToString());
-        return false;
-    }
-
-    return CheckSignature(activeMasternode.pubKeyMasternode);
-}
-
-bool CDarksendQueue::CheckSignature(const CPubKey& pubKeyMasternode)
-{
-    std::string strMessage = vin.ToString() + boost::lexical_cast<std::string>(nDenom) + boost::lexical_cast<std::string>(nTime) + boost::lexical_cast<std::string>(fReady);
-    std::string strError = "";
-
-    if(!CMessageSigner::VerifyMessage(pubKeyMasternode, vchSig, strMessage, strError)) {
-        LogPrintf("CDarksendQueue::CheckSignature -- Got bad Masternode queue signature: %s; error: %s\n", ToString(), strError);
-        return false;
-    }
-
-    return true;
-}
-
-bool CDarksendQueue::Relay()
+void CPrivateSend::AddDSTX(const CDarksendBroadcastTx& dstx)
 {
-    std::vector<CNode*> vNodesCopy = CopyNodeVector();
-    BOOST_FOREACH(CNode* pnode, vNodesCopy)
-        if(pnode->nVersion >= MIN_PRIVATESEND_PEER_PROTO_VERSION)
-            pnode->PushMessage(NetMsgType::DSQUEUE, (*this));
-
-    ReleaseNodeVector(vNodesCopy);
-    return true;
+    LOCK(cs_mapdstx);
+    mapDSTX.insert(std::make_pair(dstx.tx.GetHash(), dstx));
 }
 
```

Comment:
I was too hasty in saying remove the lock. Eating my threads.

The lock is needed if there is any possibility of writes from another thread
(thread safety of reads is only relevant for all threads reading only).
(Thanks for the prod Tim, sorry for the red herring.)

>allow to have many mixing sessions in parallel both on the server and on the client side.

Are mixing sessions currently dispatched on multiple threads, or is that the plan?
If all mixing sessions can be scheduled on one thread then the lock still isn't needed.
How much work is involved in mixing, compared to the tx handling for instance?